node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java

```diff
@@ -265,6 +276,17 @@ private Allocation requireAllocation(String message) {
         return allocation.get();
     }
 
+    /** Return the current port allocator for this node, if any */
+    public Optional<NetworkPorts> networkPorts() { return networkPorts; }
+
+    /** Return a network port allocator for this node */
+    public NetworkPorts getNetworkPorts() {
```

Comment:
Is this used? In any case, I think it's better for the caller to deal with constructing NetworkPorts if there is none.